### PR TITLE
Fix installer integration tests.

### PR DESCRIPTION
**Pull request by mitchell-as**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-850" title="DX-850" target="_blank">BugDX-850</a>  Installer Integration Test Failure (Regression)
  </td></table>
  <br />
 

This is blocking a number of my PRs.

### CODE REVIEW

File: cmd/state-installer/test/integration/installer_int_test.go [hunk 1]
```diff
@@ -35,9 +35,8 @@ func (suite *InstallerIntegrationTestSuite) TestInstallFromLocalSource() {
 		e2e.AppendEnv(constants.DisableUpdates+"=false"))
 
 	// Assert output
-	cp.Expect("Installing State Tool")
+	cp.Expect("Installing Update")
```
Comment (Naatan) [inline cmd/state-installer/test/integration/installer_int_test.go:38]: This is wrong; the test is testing a fresh install, not a update. The fact that this is failing suggests either fresh installs are giving the wrong UX or the test is mocked wrong (eg. targeting an install dir that's already being used).

**Review by Naatan — CHANGES_REQUESTED** [review]

File: cmd/state-installer/test/integration/installer_int_test.go [hunk 2]
```diff
@@ -31,13 +31,12 @@ func (suite *InstallerIntegrationTestSuite) TestInstallFromLocalSource() {
 	// Run installer with source-path flag (ie. install from this local path)
 	cp := ts.SpawnCmdWithOpts(
 		ts.InstallerExe,
-		e2e.WithArgs(target, "--source-path", ts.Dirs.Bin),
+		e2e.WithArgs(target, "--source-path", ts.Dirs.Bin, "--from-deferred"),
```
Comment (Naatan) [inline cmd/state-installer/test/integration/installer_int_test.go:34]: Note this is again working around the actual failure. It doesn't look like this test is broken, it looks like what it is testing is.

The target location should be empty. It seems for some reason the installer is detecting that the target is non-empty and that the State Tool is already installed there, and giving the update flow as a result.

Comment (mitchell-as) [reply cmd/state-installer/test/integration/installer_int_test.go:34]: If you know what to do, feel free to grab this PR or open a separate one. I'm juggling between a lot of tasks right now and am spinning my wheels on this one.

Comment (Naatan) [reply cmd/state-installer/test/integration/installer_int_test.go:34]: Ok, just unstart it on Jira so someone else can take it.

**Comment by mitchell-as** [comment]
@Naatan ping. This is blocking 3 PRs.

**Review by Naatan — CHANGES_REQUESTED** [review]
I've filed the following story to follow up on this, as this PR made me realize our installer tests are a bit "broken" regardless: https://activestatef.atlassian.net/browse/DX-869

But if this works an unblocks our tests lets merge this. Fixing this properly will take more time, and despite this being error-prone it evidently isn't causing any detectable errors atm.

**Review by Naatan — APPROVED** [review]
Meant to approve.

**Pull request merged**